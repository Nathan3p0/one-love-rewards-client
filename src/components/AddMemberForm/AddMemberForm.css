.dashboard__add-member--form {
    color: #FDFFFC;
}