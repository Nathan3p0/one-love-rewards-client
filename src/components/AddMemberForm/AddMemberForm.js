import React from 'react'
import { Link } from 'react-router-dom'

const AddMemberForm = (props) => {
    return (
        <form onSubmit={props.handleSubmit} className="dashboard__add-member--form">
            <h2>Add New Member</h2>
            <div role='alert'>
                {props.error && <p className='red'>{props.error.detail}</p>}
            </div>
            <div>
                <label htmlFor="name">Customer Name:</label>
                <br />
                <input type="text" id="name" name="name" placeholder="Full Name" onChange={props.handleInputChange} required />
            </div>
            <div>
                <label htmlFor="email">Customer Email:</label>
                <br />
                <input type="email" id="email" name="email" placeholder="Email" onChange={props.handleInputChange} required />
            </div>
            <div>
                <label htmlFor="customer-phone">Customer Phone Number:</label>
                <br />
                <input type="tel" id="customer-phone" name="phone_number" pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}" onChange={props.handleInputChange}
                    placeholder="734-555-0909" required />
            </div>
            <button type="submit" className="create-btn" >Create User</button>
            <Link to='/dashboard'><button className="go-back-btn">Go Back</button></Link>
        </form>
    )
}

export default AddMemberForm