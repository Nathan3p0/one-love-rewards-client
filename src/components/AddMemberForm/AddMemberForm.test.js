import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import AddMemberForm from './AddMemberForm'

describe('Add Points Form Component Rendering Tests', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<AddMemberForm />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders to UI as expected', () => {
        const tree = renderer.create(<AddMemberForm />).toJSON();
        expect(tree).toMatchSnapshot()
    })

    it('renders to UI as expected with PROPS', () => {
        const tree = renderer.create(<AddMemberForm
            error={'error'}
            handleInputChange={() => {}}
            handleSubmit={() => {}}
             />).toJSON();
        expect(tree).toMatchSnapshot()
    })

})