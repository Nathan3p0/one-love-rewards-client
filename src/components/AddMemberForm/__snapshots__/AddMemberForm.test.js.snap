// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Points Form Component Rendering Tests renders to UI as expected 1`] = `
<form
  className="dashboard__add-member--form"
>
  <h2>
    Add New Member
  </h2>
  <div
    role="alert"
  />
  <div>
    <label
      htmlFor="name"
    >
      Customer Name:
    </label>
    <br />
    <input
      id="name"
      name="name"
      placeholder="Full Name"
      required={true}
      type="text"
    />
  </div>
  <div>
    <label
      htmlFor="email"
    >
      Customer Email:
    </label>
    <br />
    <input
      id="email"
      name="email"
      placeholder="Email"
      required={true}
      type="email"
    />
  </div>
  <div>
    <label
      htmlFor="customer-phone"
    >
      Customer Phone Number:
    </label>
    <br />
    <input
      id="customer-phone"
      name="phone_number"
      pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}"
      placeholder="734-555-0909"
      required={true}
      type="tel"
    />
  </div>
  <button
    className="create-btn"
    type="submit"
  >
    Create User
  </button>
</form>
`;

exports[`Add Points Form Component Rendering Tests renders to UI as expected with PROPS 1`] = `
<form
  className="dashboard__add-member--form"
  onSubmit={[Function]}
>
  <h2>
    Add New Member
  </h2>
  <div
    role="alert"
  >
    <p
      className="red"
    />
  </div>
  <div>
    <label
      htmlFor="name"
    >
      Customer Name:
    </label>
    <br />
    <input
      id="name"
      name="name"
      onChange={[Function]}
      placeholder="Full Name"
      required={true}
      type="text"
    />
  </div>
  <div>
    <label
      htmlFor="email"
    >
      Customer Email:
    </label>
    <br />
    <input
      id="email"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      required={true}
      type="email"
    />
  </div>
  <div>
    <label
      htmlFor="customer-phone"
    >
      Customer Phone Number:
    </label>
    <br />
    <input
      id="customer-phone"
      name="phone_number"
      onChange={[Function]}
      pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}"
      placeholder="734-555-0909"
      required={true}
      type="tel"
    />
  </div>
  <button
    className="create-btn"
    type="submit"
  >
    Create User
  </button>
</form>
`;
